shared/actions/signup/index.js

```diff
@@ -242,8 +242,12 @@ function signup (skipMail): TypedAsyncAction<Signup | ShowPaperKey> {
   })
 }
 
-export function resetSignup (): ResetSignup {
-  return {type: Constants.resetSignup, payload: {}}
+export function resetSignup (): TypedAsyncAction<ResetSignup | RouteAppend> {
+  return dispatch => new Promise((resolve, reject) => {
+    dispatch({type: Constants.resetSignup, payload: {}})
+    dispatch(nextPhase())
```

Comment:
agree with @chrisnojima that if we can find a way to e.g. subscribe to phase changes to fire nextPhase() for us, it's better to dispatch once here instead of twice.

(I'll go ahead and merge this in though!)